### PR TITLE
[fuzz] OSS Fuzz Issue 30119: envoy:server_fuzz_test: ASSERT: not reached

### PR DESCRIPTION
**Pull request by twghu**
Signed-off-by: Tim Walsh <temporal.differential@gmail.com>

Commit Message: Update for oss-fuzz Issue 30119
Additional Description: Note that #4709 covers a number of oss-fuzz error This is part of the ongoing effort started for #4709.
Risk Level: Low
Testing: test/common/stats
Docs Changes: None
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional API Considerations:]

### CODE REVIEW

**Comment by phlax** [comment]
@twghu can you fix DCO please

**Review by jmarantz — COMMENTED** [review]
Looks fine for correctness but I'm wondering if this validation prevents most fuzz input from reaching anything.

Can you comment on the possibility of using the fuzzing input to generate valid results?

**Comment by twghu** [comment]
The apt_type values for [config.core.v3.ApiConfigSource](https://www.envoyproxy.io/docs/envoy/v1.18.3/api-v3/config/core/v3/config_source.proto.html?highlight=api_type) only include the three types when loaded from file.  Testing with dynamic-config-fs example follows a validation path that is not traversed from the test case.

Specifically, in the xds case, with api boost active, the validation correctly identifies errors:
```
[libprotobuf ERROR external/com_google_protobuf/src/google/protobuf/text_format.cc:321] Error parsing text-format envoy.api.v2.DiscoveryResponse: 1:19: Message type "envoy.api.v2.DiscoveryResponse" has no field named "dynamic_resources".
[libprotobuf ERROR external/com_google_protobuf/src/google/protobuf/text_format.cc:321] Error parsing text-format envoy.service.discovery.v3.DiscoveryResponse: 1:19: Message type "envoy.service.discovery.v3.DiscoveryResponse" has no field named "dynamic_resources".
[2021-06-08 20:30:56.903][140670][warning][config] [source/common/config/filesystem_subscription_impl.cc:81] Filesystem config update failure: Unable to parse file "./configs/cds.pb_text" as a text protobuf (type envoy.service.discovery.v3.DiscoveryResponse)
``` 

Given that testing with a corpus of some 1300 entries yielded only 2 hits on the validation exclusion added in this PR, and also according to documentation [1],[2] regarding acceptable values for `api_tpe` it would seem reasonable use `validateDynamicResources()` in this test.  Ideally though, a consistent validation path is ideal but in the test case the use of \*.pb_text appears to bypass validation in 
```c++
oid MessageUtil::loadFromFile(const std::string& path, Protobuf::Message& message,
                               ProtobufMessage::ValidationVisitor& validation_visitor,
                               Api::Api& api, bool do_boosting)
```

In conclusion:
1) Accept the validateDynamicResources() modification or
2) Add cds validation logic via the CdsApiImpl code pathway.

[1] - https://www.envoyproxy.io/docs/envoy/v1.18.3/configuration/overview/xds_api.html?highlight=api_type
[2] - https://www.envoyproxy.io/docs/envoy/v1.18.3/configuration/overview/examples.html?highlight=api_type
[3] - https://www.envoyproxy.io/docs/envoy/v1.18.3/api-v3/config/core/v3/config_source.proto.html?highlight=api_type
[4] - https://www.envoyproxy.io/docs/envoy/v1.18.3/start/quick-start/configuration-dynamic-control-plane.html?highlight=api_type

**Comment by htuch** [comment]
Not sure why API boost is relevant anymore, since we should be v3 only. Can we fuzz just v3 configs?
